### PR TITLE
[OT210-34] Update Endpoint User created

### PR DESCRIPTION
**Pull request by hernancostarelli**
### Motivation and Summary

Descripción

PATCH a /users/:id. En el caso de que el usuario no exista, devolverá un error 404.

### Type of change
- [x] New feature
- [ ] New Tests
- [ ] Bug fix
- [x] Refactor

### Checklist

- [x] Traceability between this change and Jira.
- [x] ```mvn clean install``` was run and all tests completed successfully.
- [x] There are no unused variables and/or imports in the modified classes.
- [x] There are no imports in the modified classes with the wildcard character. Ex: ```com.somepackage.*```.
- [ ] Slf4j was used for the application log instead ```System.out```.
- [ ] Public methods are documented with ```javadoc```.

### CODE REVIEW

File: src/main/java/com/alkemy/ong/domain/repository/UserRepository.java [hunk 1]
```diff
@@ -7,4 +7,6 @@
 
 public interface UserRepository extends PagingAndSortingRepository<User, Long> {
     Optional<User> findByEmail(String email);
+
+    User getById(Long id);
```
Comment (manerajona) [inline src/main/java/com/alkemy/ong/domain/repository/UserRepository.java:11]: No hace falta crear este método, usar `findById`.

File: src/main/java/com/alkemy/ong/domain/usecase/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -29,6 +39,38 @@ public UserDetails loadUserByUsername(String email) {
...
-        return new UserList(page.getContent(),pageRequest,page.getTotalElements());
+        return new UserList(page.getContent(), pageRequest, page.getTotalElements());
+    }
+
+    @Override
+    @Transactional
+    public UserResponse updateUser(Long id, User entity) {
+
+        try {
+            User userToUpdate = userJpaRepository.getById(id);
```
Comment (manerajona) [inline src/main/java/com/alkemy/ong/domain/usecase/impl/UserServiceImpl.java:50]: Usar el método findById que devuelve un optional para validar la existencia y determinar si se tiene que devolver un `404`.

```java
Optional<User> op = userJpaRepository.findById(id);
if(op.isPresent()) {
  // actualizar
} else {
  throw new NotFoundException(id);
}
```

**Review by manerajona — CHANGES_REQUESTED** [review]

**Comment by manerajona** [comment]
@hernancostarelli te quedó con conflictos el pr

**Comment by hernancostarelli** [comment]
- requested changes made

**Pull request merged**